src/tests/ftest/server/multiengine_persocket.py

```diff
@@ -246,6 +247,7 @@ def test_multiengines_per_socket(self):
             # Check for agent start status
             if results[0]['exit_status'] and "sudo systemctl" in cmd:
                 self.fail("#Fail on {0}".format(cmd))
+        time.sleep(15)
```

Comment:
this should be waiting for ranks to join in a deterministic way, not just waiting for a static length of time, as it is this is brittle and may break (there is no guarantee that the ranks will join within this time)

Reply:
Thanks for the review, 
the reason to use a fixed delay is due to after "daos_server scm prepare --force --scm-ns-per-socket=2", and "dmg storage format", observing 2 server-ranks joined, and some time 4 server-ranks joined.

Reply:
right so I am suggesting that you wait until 4 ranks have joined before continuing by looping over the results of system query and not continuing until 4 ranks are detected

Reply:
Issue is script configure server with "--scm-ns-per-socket=2", but detected with 2 ranks or 4 ranks up occasionally,.

Reply:
@dinghwah : So you are saying sometimes only 2 ranks are detected (however long you wait?)... If so, is it not a bug?

Reply:
This should always be 4, I presume this is because of the issue where namepaces are created with unexpected names e.g.  0, 0.2, 1, 1.2 instead of the usual 0, 0.1, 1, 1.1. @dinghwah Could you please try with the changes I've suggested above and see if the same issue occurs?